core/src/main/java/eu/xenit/alfresco/AntiIdleScheduledJobExecuter.java

```diff
@@ -14,12 +41,129 @@
 
     public void setServiceRegistry(ServiceRegistry serviceRegistry) {
         this.serviceRegistry = serviceRegistry;
+        this.nodeService = serviceRegistry.getNodeService();
+        this.nodeLocatorService = serviceRegistry.getNodeLocatorService();
+        this.retryingTransactionHelper = serviceRegistry.getRetryingTransactionHelper();
+    }
+
+    public void setFileLocation(String location) {
+        if (StringUtils.isEmpty(location)) {
+            LOG.debug("Custom location not set, using default " + defaultLocation);
+            setFolderAndFileName(toSegments(defaultLocation));
+            return;
+        }
+        List<String> segments = toSegments(location);
+        if (segments.isEmpty()) {
+            LOG.debug("Custom location should have at least one segment, using default " + defaultLocation);
+            setFolderAndFileName(toSegments(defaultLocation));
+            return;
+        }
+        LOG.debug("Setting location to " + location);
+        setFolderAndFileName(segments);
+    }
+
+    private void setFolderAndFileName(List<String> segments) {
```

Comment:
A method like that is pretty ugly: the name says it's about setting folder and filename, but it's doing some operations on a list of strings.

Reply:
and mutating the list

Reply:
updated to not mutate the list